src/Passes/Source/ModuleLoader/ModuleLoader.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "RemoveDisallowedAttribubtesPass/RemoveDisallowedAttribubtesPass.hpp"
+#include "Types/Types.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class SingleModuleTransformation
+    {
+      public:
+        using PassBuilder             = llvm::PassBuilder;
+        using OptimizationLevel       = PassBuilder::OptimizationLevel;
+        using FunctionAnalysisManager = llvm::FunctionAnalysisManager;
+
+        explicit SingleModuleTransformation(
+            OptimizationLevel const& optimisation_level = OptimizationLevel::O0,
+            bool                     debug              = false)
+          : loop_analysis_manager_{debug}
+          , function_analysis_manager_{debug}
+          , gscc_analysis_manager_{debug}
+          , module_analysis_manager_{debug}
+          , optimisation_level_{optimisation_level}
+          , debug_{debug}
+        {
+
+            pass_builder_.registerModuleAnalyses(module_analysis_manager_);
+            pass_builder_.registerCGSCCAnalyses(gscc_analysis_manager_);
+            pass_builder_.registerFunctionAnalyses(function_analysis_manager_);
+            pass_builder_.registerLoopAnalyses(loop_analysis_manager_);
+
+            pass_builder_.crossRegisterProxies(
+                loop_analysis_manager_, function_analysis_manager_, gscc_analysis_manager_, module_analysis_manager_);
+
+            module_pass_manager_.addPass(RemoveDisallowedAttribubtesPass());
+        }
+
+        bool apply(llvm::Module* module)
+        {
+            module_pass_manager_.run(*module, module_analysis_manager_);
+
+            if (verifyModule(*module, &llvm::errs()))
+            {
+                return false;
+            }
```

Comment:
I fail to find the definition of `verifyModule()` (with at least 2 parameters).

Reply:
It is a LLVM function: https://llvm.org/doxygen/Verifier_8h.html